pkg/api/types.go

```diff
@@ -758,12 +758,12 @@ type NodeSpec struct {
 
 // NodeStatus is information about the current status of a node.
 type NodeStatus struct {
-	// Queried from cloud provider, if available.
-	HostIP string `json:"hostIP,omitempty"`
 	// NodePhase is the current lifecycle phase of the node.
 	Phase NodePhase `json:"phase,omitempty"`
 	// Conditions is an array of current node conditions.
 	Conditions []NodeCondition `json:"conditions,omitempty"`
+	// Queried from cloud provider, if available.
+	Addresses []NodeAddress `json:"addresses,omitempty"`
```

Comment:
Why Alternative in the Name?  Do we have a primary Endpoint or Address?

Endpoints is more or less a concept in our service world now, do we want to overload it here?  It's a little confusing, plus I don't know how useful it would be to have port. Whoever consumes the list should know the port, like api-server proxy.

I think what we need here is a list of equivalent address (external ip, internal ip, hostname).  If the direction is to produce Endpoints in Node, then I'll need to take a deep look at how it might work.